test/aws_secrets_provider/aws_secrets_provider_test.go

```diff
@@ -0,0 +1,30 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/cyberark/secretless-broker/internal/app/secretless/providers"
+	plugin_v1 "github.com/cyberark/secretless-broker/pkg/secretless/plugin/v1"
+	_ "github.com/joho/godotenv/autoload"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestAWSSecrets_Provider(t *testing.T) {
+	var err error
+	var provider plugin_v1.Provider
+
+	name := "aws"
+
+	options := plugin_v1.ProviderOptions{
+		Name: name,
+	}
+
+	Convey("Can create the AWS Secrets provider", t, func() {
+		provider, err = providers.ProviderFactories[name](options)
+		So(err, ShouldBeNil)
+	})
+
+	Convey("Has the expected provider name", t, func() {
+		So(provider.GetName(), ShouldEqual, "aws")
+	})
```

Comment:
This isn't a real integration test but more of a unit test and it doesn't test pulling secrets from it. Either we should actually make this do a real aws test or move this to our unit testing framework.